src/aggregation/agg_req_with_accessor.rs

```diff
@@ -54,38 +59,68 @@ impl AggregationWithAccessor {
         reader: &SegmentReader,
         limits: AggregationLimits,
     ) -> crate::Result<Vec<AggregationWithAccessor>> {
-        let mut missing_value_term_agg = None;
-        let mut str_dict_column = None;
+        let get_agg_with_accessor = |aggs: &mut Vec<AggregationWithAccessor>,
```

Comment:
does this have to be inlined? can it be a proper funciton?

Reply:
nitpick:
I suggest we try to stick to a rule for argument lists: the mut stuff at the end of the list of arguments.
This is arbitrary. (that's the rule enforced at Google)

Reply:
It sounds like it could have been a function with a clearer prototype:

`Fn(.., .., .., .., ..) -> Option<AggregationWithAccessor> {}`

Reply:
The closure has access to the parameters and is mainly there to remove code duplication, moving it to its own function has too many parameters imo

```rust
fn add_agg_with_accessor(
    aggs: &mut Vec<AggregationWithAccessor>,
    accessor: Column<u64>,
    column_type: ColumnType,
    str_dict_column: Option<StrColumn>,
    missing_value_term_agg: Option<Key>,
    agg: &Aggregation,
    sub_aggregation: &Aggregations,
    reader: &SegmentReader,
    limits: AggregationLimits,
) -> crate::Result<()> {
```

I refactored it a little bit and added types to the closure instead.


> nitpick:
I suggest we try to stick to a rule for argument lists: the mut stuff at the end of the list of arguments.
This is arbitrary. (that's the rule enforced at Google)

I'm not really in favor of that, since this would cause churn when refactoring something to mut for very little gain (at least for how I read code).
I changed it in this case though.